components/builder-api/doc/api.raml

```diff
@@ -267,6 +267,21 @@ securitySchemes:
                 description: Project does not exist with corresponding projectId
             422:
                 description: Invalid or missing projectId in body
+    /group:
+        /{id}:
+            /promote:
+                /{channel}:
+                    post:
+                        description: Promote every successful project in a job group to the specified channel
+                        responses:
+                            200:
+                                description: Successful promotion
+                            400:
+                                description: ID or channel not provided
+                            404:
+                                description: Group not found
+                            500:
+                                description: Internal server error
```

Comment:
This was mostly just a port of the functionality as it existed in the scheduler, but you bring up some good points.

* If any of the projects didn't succeed, we can return those and indicate that they failed.
* If the group itself failed, we should indicate that in the return value as well.

For both of the above items, keeping the status as 200 seems appropriate, no?

If you promote to a channel that the packages are already in, nothing happens.  It goes through like it normally would.